main.nf

```diff
@@ -20,16 +20,13 @@ nextflow.enable.dsl = 2
 include { SCRNASEQ                } from './workflows/scrnaseq'
 include { PIPELINE_INITIALISATION } from './subworkflows/local/utils_nfcore_scrnaseq_pipeline'
 include { PIPELINE_COMPLETION     } from './subworkflows/local/utils_nfcore_scrnaseq_pipeline'
-include { getGenomeAttribute      } from './subworkflows/local/utils_nfcore_scrnaseq_pipeline'
 
 /*
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     GENOME PARAMETER VALUES
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
 // we cannot modify params. here, we must load the files
-ch_genome_fasta = params.genome ? file( getGenomeAttribute('fasta'), checkIfExists: true ) : []
-ch_gtf          = params.genome ? file( getGenomeAttribute('gtf'), checkIfExists: true   ) : []
```

Comment:
Was it required to be removed from the `main.nf`. Was it it the "main"-template functions that were complaining about not providing a fasta?

If that is the case, I would say this is a ticket that should be taken entirely at once, in a separate one, because it would affect all of the workflows.

Thus having a single issue to take a look at this matter for all workflows would be easier to track.

Issues that could be optionally merged together for it are: #313 and #277 .

Reply:
I think #313 is already part of the cellranger multi PR. 

I wasn't aware that there's a linting check for that, but I'd argue that as it is currently implemented is suboptimal. Either
 * all parameters should be evaluated outside the scrnaseq workflow and passed as arguments
 * all parameters should be evaluated within the workflow and not arguments should be passed. 

Currently it is just inconsistent with fastq/gtf being passed as arguments and to me it's highly confusing that different parts of the workflow partly evaluate some parameters. It also doesn't help because you anyway can't `include` the workflow somewhere else and run it as it is currently implemented. 

For the sake of simplicitly I'd vote for sticking with what we currently have (i.e. evaluate parameters within the workflow) and if necessary ignore the respective linting checks. The solution intended by the template authors would probably be to move all parameter checks outside the workflow and pass them as arguments.

Reply:
linting doesn't seem to fail because of this btw.

Reply:
That is true, every other subworkflow currently, have the input assertion happening inside the sub-workflow itself:

e.g. alevin
```groovy
assert (genome_fasta && gtf && salmon_index && txp2gene) || (genome_fasta && gtf)  || (genome_fasta && gtf && transcript_fasta && txp2gene):
        """Must provide a genome fasta file ('--fasta') and a gtf file ('--gtf'), or a genome fasta file
        and a transcriptome fasta file ('--transcript_fasta`) if no index and txp2gene is given!""".stripIndent()
```

However, they are indeed seeming to rely on gtf/fasta that had been `file()` loaded outside. So, we should also bring the loaders to the inside the sub-workflow on the others as well.

Is that the correct understanding?

Reply:
Ah!,
Now I can see it clearly that the changes were happening in these two.
For some reason I was reading as if they were included inside the cellrangermulti sub-workflow.
🤯